### PR TITLE
Fix sign of phase shift in DataSet.apply_symop

### PR DESCRIPTION
**Pull request by JBGreisman**
Based on visual inspection of maps, it appeared that `DataSet.apply_symop()` was applying phase shifts in the opposite direction of the specified symmetry operation. This PR changes the sign for how phase shifts are applied, and updates the relevant test.

### CODE REVIEW

**Comment by JBGreisman** [comment]
To do:
 - It would be useful to make a "gold standard" pair of F/phi with a known real-space translation between the corresponding maps so that we can write a cleaner test.

**Comment by JBGreisman** [comment]
My most recent push added a test that uses fmodel data to compute a real-space map, shifts the map based on a desire translation using `np.roll`, back computes corresponding complex structure factors. These are then compared to the results of `DataSet.apply_symop()`

File: reciprocalspaceship/dataset.py [hunk 1]
```diff
@@ -597,8 +597,7 @@ def apply_symop(self, symop, inplace=False):
 
         # Shift phases according to symop
         for key in dataset.get_phase_keys():
-            dataset[key] += np.rad2deg(phase_shifts)
-            dataset[key] *= phic
+            dataset[key] = phic * (dataset[key] - np.rad2deg(phase_shifts))
             dataset[key] = utils.canonicalize_phases(dataset[key], deg=True)
         for key in dataset.get_complex_keys():
             dataset[key] *= np.exp(1j*phase_shifts)
```
Comment (kmdalton) [inline reciprocalspaceship/dataset.py:603]: Don't you need to change the sign of this complex rotation too?

**Comment by kmdalton** [comment]
As per our conversation offline, let's go ahead and merge this and then do a refactor.

**Review by kmdalton — CHANGES_REQUESTED** [review]
I am not clear why you did not change the sign of the phase shift applied to complex structure factors. Not sure if this is in error. If tests are passing, I am concerned that complex phase shifts are maybe not adequately tested. 

Can you either change the sign of complex phase shifts or tell me that there is no need for some reason that I am missing?

**Comment by JBGreisman** [comment]
I am still working on this for the complex structure factors. This test may not be working as well as I had hoped, but I am not yet ready to come to that conclusion

**Comment by JBGreisman** [comment]
This PR now fixes `DataSet.apply_symop()` for use with both |F|+phi and complex SFs. I updated the test to use a P1 dataset avoid an interdependency caused by the use of `apply_symop()` in `DataSet.to_reciprocalgrid()`. 

P.S. -- I updated my emacs recently to auto-apply `black` formatting, which is why there are additional changes to the `test_dataset_symops.py` file

File: reciprocalspaceship/dataset.py (hunk 1 above)
Comment (JBGreisman) [reply reciprocalspaceship/dataset.py]: Fixed in latest update -- the phase shifts have now been flipped for complex structure factors as well

File: tests/test_dataset.py [hunk 2]
```diff
@@ -495,14 +495,14 @@ def test_apply_symop_roundtrip(mtz_by_spacegroup):
     dataset = rs.read_mtz(mtz_by_spacegroup)
     for op in dataset.spacegroup.operations():
         applied = dataset.apply_symop(op)
-        back = applied.hkl_to_asu()
```
Comment (kmdalton) [inline tests/test_dataset.py:498]: We should try to resurrect this test when we do the refactor, but this is okay with me for now.

**Review by kmdalton — APPROVED** [review]
This looks good to me.

**Pull request merged**